modules/@ergonode/core/src/components/Forms/MainSettingsForm.vue

```diff
@@ -39,32 +50,43 @@ export default {
     name: 'MainSettingsForm',
     components: {
         InfoHint,
-        Form: () => import('@Core/components/Form/Form'),
-        FormSection: () => import('@Core/components/Form/Section/FormSection'),
-        TranslationSelect: () => import('@Core/components/Inputs/Select/TranslationSelect'),
+        Form,
+        FormSection,
+        TranslationSelect,
     },
+    mixins: [
+        formActionsMixin,
+    ],
     data() {
         return {
             filteredValue: '',
-            tmpLanguages: [],
             activeLanguages: [],
         };
     },
     computed: {
         ...mapState('core', {
             languages: state => state.languages,
         }),
+        mappedLanguages() {
+            return this.languages.map(({
+                id, name, code,
+            }) => ({
+                id,
+                key: code,
+                value: name,
+            }));
+        },
```

Comment:
maybe move this mapper to models section, we should keep all out mappers in one place.